sdk/containerregistry/azure-containers-containerregistry/src/main/java/com/azure/containers/containerregistry/specialized/ContainerRegistryBlobAsyncClient.java

```diff
@@ -0,0 +1,468 @@
+    public Mono<Response<Void>> deleteBlobWithResponse(String digest) {
+        return withContext(context -> deleteBlobWithResponse(digest, context));
+    }
+
+    Mono<Response<Void>> deleteBlobWithResponse(String digest, Context context) {
+        if (digest == null) {
+            return monoError(logger, new NullPointerException("'digest' can't be null."));
+        }
+
+        return this.blobsImpl.deleteBlobWithResponseAsync(repositoryName, digest, context)
+            .flatMap(UtilsImpl::deleteResponseToSuccess)
+            .onErrorMap(UtilsImpl::mapException);
+    }
+
+    /**
+     * Delete the manifest associated with the given digest.
+     * We currently only support downloading OCI manifests.
+     *
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     *
+     * @param digest The digest of the manifest.
+     * @return The completion.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code digest} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Void> deleteManifest(String digest) {
+        return this.deleteManifestWithResponse(digest).flatMap(FluxUtil::toMono);
+    }
+
+    /**
+     * Delete the manifest associated with the given digest.
+     * We currently only support downloading OCI manifests.
+     *
+     * @see <a href="https://github.com/opencontainers/image-spec/blob/main/manifest.md">Oci Manifest Specification</a>
+     *
+     * @param digest The digest of the manifest.
+     * @return The REST response for completion.
+     * @throws ClientAuthenticationException thrown if the client's credentials do not have access to modify the namespace.
+     * @throws NullPointerException thrown if the {@code digest} is null.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<Response<Void>> deleteManifestWithResponse(String digest) {
+        return withContext(context -> deleteManifestWithResponse(digest, context));
+    }
+
+    Mono<Response<Void>> deleteManifestWithResponse(String digest, Context context) {
+        return this.registriesImpl.deleteManifestWithResponseAsync(repositoryName, digest, context)
+            .flatMap(UtilsImpl::deleteResponseToSuccess)
+            .onErrorMap(UtilsImpl::mapException);
+    }
+
+    static final class BufferedFlux {
+        int size;
+        ByteBuffer byteBuffer;
+
+        int getSize() {
+            return this.size;
+        }
+
+        String getDigest() {
+            return UtilsImpl.computeDigest(byteBuffer);
+        }
+
+        Flux<Void> write(ByteBuffer buffer) {
+            size += buffer.remaining();
+            byteBuffer = buffer;
+            return Flux.empty();
+        }
+
+        Flux<ByteBuffer> flush() {
+            return Flux.just(byteBuffer);
+        }
+    }
```

Comment:
This may not be required if we directly use `ByteBuffer` instead of converting user input to `Flux<ByteBuffer>`